packages/lix-sdk/src/branch/merge-branch.ts

```diff
@@ -0,0 +1,171 @@
+import type { Branch, Change } from "../database/schema.js";
+import type { Lix } from "../open/openLix.js";
+import type { DetectedConflict } from "../plugin/lix-plugin.js";
+import { getLowestCommonAncestorV2 } from "../query-utilities/get-lowest-common-ancestor-v2.js";
+
+export async function mergeBranch(args: {
+	lix: Lix;
+	sourceBranch: Branch;
+	targetBranch: Branch;
+}) {
+	const executeInTransaction = async (trx: Lix["db"]) => {
+		const diffingPointers = await getBranchChangePointerDiff(trx, args);
+
+		// could be queried in a single query with a join
+		// but decided to keep it simpler for now
+		const diffingChanges = await trx
+			.selectFrom("change")
+			.select(["id"])
+			.where(
+				"id",
+				"in",
+				diffingPointers.map((pointer) => pointer.source_change_id),
+			)
+			.selectAll()
+			.execute();
+
+		const plugins = await args.lix.plugin.getAll();
+		const detectedConflicts: DetectedConflict[] = [];
+
+		// let plugin detect conflicts
+		await Promise.all(
+			plugins.map(async (plugin) => {
+				if (plugin.detectConflictsV2) {
+					const conflicts = await plugin.detectConflictsV2({
+						lix: args.lix,
+						changes: diffingChanges,
```

Comment:
don't we need to provide the information about the target branch as well?

Reply:
i thought so too but it doesn't seem like it. 

- a branch points to a set of changes 
- merging two branches leads to a new set 
- a plugin only needs to detect if a change in the new set exists 

A nice side effect, the API generalizes pretty well. This API can now be used to detect conflicts even within the same branch (because it's just a set) or other domains.